go/private/toolchain.bzl

```diff
@@ -12,24 +12,67 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-def _go_sdk_repository_impl(ctx):
-  goroot = ctx.path(".")
-  ctx.template("BUILD.bazel", 
-    Label("@io_bazel_rules_go//go/private:BUILD.sdk.bazel"),
-    substitutions = {"{goroot}": str(goroot)}, 
-    executable = False,
-  )
-  ctx.download_and_extract(
-      url = ctx.attr.url,
-      stripPrefix = ctx.attr.strip_prefix,
-      sha256 = ctx.attr.sha256)
+def _go_host_sdk_impl(ctx):
+  root = _detect_host_sdk(ctx)
+  _local_sdk(ctx, root)
+  _skd_build_file(ctx, root)
+
+go_host_sdk = repository_rule(_go_host_sdk_impl, environ = ["GOROOT"])
+"""go_host_sdk is a specialization of go_sdk just to add the GOROOT dependancy"""
 
+def _go_sdk_impl(ctx):
+  _skd_build_file(ctx, str(ctx.path(".")))
+  if ctx.attr.url:
+    if ctx.attr.root:
+      fail("url and root cannot both be set on go_sdk, got {} and {}".format(ctx.attr.url, ctx.attr.root))
+    _remote_sdk(ctx, ctx.attr.url, ctx.attr.strip_prefix, ctx.attr.sha256)
+  elif ctx.attr.root:
+    _local_sdk(ctx, ctx.attr.root)
+    _skd_build_file(ctx, ctx.attr.root)
+  else:
+    root = _detect_host_sdk(ctx)
+    _local_sdk(ctx, root)
+    _skd_build_file(ctx, root)
+    
   # Build the standard library for valid cross compile platforms
+  #TODO: fix standard library cross compilation
   if ctx.name.endswith("linux_amd64") and ctx.os.name == "linux":
     _cross_compile_stdlib(ctx, "windows", "amd64")
   if ctx.name.endswith("darwin_amd64") and ctx.os.name == "mac os x":
     _cross_compile_stdlib(ctx, "linux", "amd64")
 
+go_sdk = repository_rule(
+    implementation = _go_sdk_impl, 
+    attrs = {
+        "root" : attr.string(),
```

Comment:
I think this should be called `path` instead of `root` for consistency with `local_repository`.

Also, micro-nit: no space before `:`

Reply:
Done